File: patch_map.py
import re

with open('src/components/Map.jsx', 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Remove getUniqueCompetitorCategories
content = re.sub(r'const getUniqueCompetitorCategories = .*?^    };\n', '', content, flags=re.MULTILINE|re.DOTALL)

# 2. Remove getGroupedCompetitorData
content = re.sub(r'const getGroupedCompetitorData = .*?^    };\n', '', content, flags=re.MULTILINE|re.DOTALL)

# 3. Remove parseCSVToGeoJSON
content = re.sub(r'const parseCSVToGeoJSON = .*?^    };\n', '', content, flags=re.MULTILINE|re.DOTALL)

# 4. Remove parseclientCSV
content = re.sub(r'const parseclientCSV = .*?^    };\n', '', content, flags=re.MULTILINE|re.DOTALL)

# 5. Remove parseCompetitorCSV
content = re.sub(r'const parseCompetitorCSV = .*?^    };\n', '', content, flags=re.MULTILINE|re.DOTALL)

# 6. Remove the data useEffects
content = re.sub(r'// Parse hexagon data only once when it changes.*?^    }, \[competitorData\]);\n', '', content, flags=re.MULTILINE|re.DOTALL)

# 7. Replace competitorData usage
content = content.replace('competitorData && (() => {', 'competitorSummary && competitorSummary.grouped && (() => {')
content = content.replace('if (competitorData) {', 'if (competitorSummary && competitorSummary.grouped) {')

# 8. Replace getGroupedCompetitorData calls
content = content.replace('const groups = getGroupedCompetitorData(competitorData, selectedRegions);', 'const groups = competitorSummary.grouped;')

# 9. In useEffect for fetchData
content = content.replace("map.current.on('load', fetchData);", """
        map.current.on('load', fetchData);
    }, []);

    // Fetch data whenever selectedRegions changes
    useEffect(() => {
        fetchData();
    }, [selectedRegions]);

    // Dummy line to match
""")

# Fix the duplicate block introduced by 9 if any (just simple replace)
with open('src/components/Map.jsx', 'w', encoding='utf-8') as f:
    f.write(content)

print("success")

File: patch_map.js
const fs = require('fs');

let content = fs.readFileSync('src/components/Map.jsx', 'utf-8');

function removeBlock(startStr, endStr) {
    let startIndex = content.indexOf(startStr);
    while (startIndex !== -1) {
        let endIndex = content.indexOf(endStr, startIndex);
        if (endIndex !== -1) {
            content = content.substring(0, startIndex) + content.substring(endIndex + endStr.length);
        } else {
            break;
        }
        startIndex = content.indexOf(startStr);
    }
}

removeBlock('const getUniqueCompetitorCategories = (csvData) => {', '    };\n\n    const getGroupedCompetitorData');
removeBlock('const getGroupedCompetitorData = (csvData, currentSelectedRegions) => {', '    };\n\n    const toggleGroupAccordion');

removeBlock('const parseCSVToGeoJSON = (csvData) => {', '    const parseclientCSV = ');
removeBlock('const parseclientCSV = (csvData) => {', '    const parseCompetitorCSV = ');
removeBlock('const parseCompetitorCSV = (csvData) => {', '    // Add this useEffect after your existing dropdown close effect');

removeBlock('// Parse hexagon data only once when it changes', '    // Add this useEffect after your existing dropdown close effect');

// Replacing competitorData with competitorSummary.grouped
content = content.replace(/competitorData && \(\(\) => \{/g, 'competitorSummary && competitorSummary.grouped && (() => {');
content = content.replace(/if \(competitorData\) \{/g, 'if (competitorSummary && competitorSummary.grouped) {');
content = content.replace(/const groups = getGroupedCompetitorData\(competitorData, selectedRegions\);/g, 'const groups = competitorSummary.grouped;');


fs.writeFileSync('src/components/Map.jsx', content, 'utf-8');
console.log('Patch applied successfully');
